src/Apps/ArtAdvisor/04-Bio-Follows-Alerts/server.ts

```diff
@@ -0,0 +1,241 @@
+import OpenAI from "openai"
+import express, { Request, Response } from "express"
+import { instructions } from "Apps/ArtAdvisor/04-Bio-Follows-Alerts/lib/system-instructions"
+import chalk from "chalk"
+import {
+  createAlert,
+  followArtist,
+  getUserProfile,
+  updateCollectorProfile,
+} from "Apps/ArtAdvisor/04-Bio-Follows-Alerts/lib/functions"
+
+/*
+ * Setup express server, OpenAi client, and config.
+ */
+
+const openai = new OpenAI() // Client will look for OPENAI_API_KEY in your environment during instantiation
+
+/*
+ * Define the tools that the assistant can use to make function calls.
+ */
+
+const tools: OpenAI.Chat.Completions.ChatCompletionTool[] = [
+  {
+    type: "function",
+    function: {
+      name: "get_user_profile",
+      description: `Get information associated with a user profile on artsy.`,
+      parameters: {
+        type: "object",
+        properties: {
+          token: {
+            type: "string",
+            description:
+              "user's access token, which can be used to fetch user information from artsy.",
+          },
+        },
+      },
+    },
+  },
+  {
+    type: "function",
+    function: {
+      name: "update_collector_profile",
+      description: `Update the user's collector profile on artsy.`,
+      parameters: {
+        type: "object",
+        properties: {
+          bio: {
+            type: "string",
+            description:
+              "The final version of the collector profile to be saved to the user's profile.",
+          },
+        },
+      },
+    },
+  },
+  {
+    type: "function",
+    function: {
+      name: "follow_artist",
+      description: `Follow an artist on artsy on behalf of a user.`,
+      parameters: {
+        type: "object",
+        properties: {
+          artistID: {
+            type: "string",
+            description:
+              "The ID of the artist to follow. It is the same as the artsy.net slug",
+          },
+        },
+      },
+    },
+  },
+  {
+    type: "function",
+    function: {
+      name: "create_alert",
+      description: `Create an alert for an user on artsy.`,
+      parameters: {
+        type: "object",
+        properties: {
+          artistID: {
+            type: "string",
+            description:
+              "The ID of the artist to follow. It is the same as the artsy.net slug. This parameter is required.",
```

Comment:
**suggestion (non-blocking):** I don't think this was a factor in the failure to create alerts, but maybe…

```suggestion
              "The ID of the artist for whom the alert is being set. It is the same as the artsy.net slug. This parameter is required.",
```